test-packages/integration-tests/test/request-builder-v4.spec.ts

```diff
@@ -0,0 +1,122 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+
+import jwt from 'jsonwebtoken';
+import nock from 'nock';
+import { TestEntity } from '@sap-cloud-sdk/core/test/test-util/test-services/v4/test-service';
+import { destinationServiceUri } from '../../../packages/core/test/test-util/environment-mocks';
+import { privateKey } from '../../../packages/core/test/test-util/keys';
+import { basicCredentials } from './test-util/destination-encoder';
+import { testEntityCollectionResponse } from './test-data/test-entity-collection-response-v4';
+
+const servicePath = '/sap/opu/odata/sap/API_TEST_SRV';
+const entityName = 'A_TestEntity';
+const username = 'username';
+const password = 'password';
+const url = 'https://example.com';
+
+const getAllResponse = testEntityCollectionResponse();
+
+const providerToken = jwt.sign({ zid: 'provider_token' }, privateKey(), {
+  algorithm: 'RS512'
+});
+
+let destination;
+
+describe('Request Builder', () => {
+  beforeEach(() => {
+    delete process.env.destinations;
+    delete process.env.VCAP_SERVICES;
+
+    destination = {
+      url,
+      username,
+      password,
+      sapClient: '123',
+      authTokens: [],
+      originalProperties: {}
+    };
+  });
+
+  afterEach(() => {
+    nock.cleanAll();
+  });
+
+  it('should resolve for getAll request', async () => {
+    nock(destination.url, {
+      reqheaders: {
+        authorization: basicCredentials(destination),
+        accept: 'application/json',
+        'content-type': 'application/json'
+      }
+    })
+      .get(`${servicePath}/${entityName}?$format=json`)
+      .reply(200, getAllResponse);
+
+    const request = TestEntity.requestBuilder().getAll().execute(destination);
+
+    await expect(request).resolves.not.toThrow();
+  });
+
+  it('should resolve for all field', async () => {
+    nock(destination.url, {
+      reqheaders: {
+        authorization: basicCredentials(destination),
+        accept: 'application/json',
+        'content-type': 'application/json'
+      }
+    })
+      .get(`${servicePath}/${entityName}?$format=json&$select=*`)
```

Comment:
Minor: I think it is generally a better approach to use the object style notation for query strings (https://github.com/nock/nock#specifying-request-query-string), because it makes easier to account for order and encoding. I'll leave this up to you ;)